### DISCUSSION TITLE
Change play music from an Artist to not shuffle

### DISCUSSION THREAD

**Discussion by Panoramiac** [music-assistant/support#2873] [Feature requests and ideas]
### What version of Music Assistant has the issue?

2.2.2

### What version of the Home Assistant Integration have you got installed?

2024.8.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When I start playing an Artist from my local network source, I expect to get all tracks in the order of the Album releases and then track order per Album in the queue. Shuffle is disabled for my Denon receiver (using DLNA). 

![Screenshot_20240831_161547_Home Assistant](https://github.com/user-attachments/assets/acd1bca5-e50e-4f52-899c-f084126de459)
![Screenshot_20240831_161706_Home Assistant](https://github.com/user-attachments/assets/54f078ab-2c93-4260-bfbb-3b64756d49be)

Even sorting the Albums view in the Artist folder to the release order will not work. Sorting the track view by Albums will not sort the tracks  by the release date of the Albums. High as Hope it the newest Album and must be at the end of the list:

![Screenshot_20240831_162132_Home Assistant](https://github.com/user-attachments/assets/daaf64fc-54a6-48a4-8cbe-08cf6562a5b0)
![Screenshot_20240831_162138_Home Assistant](https://github.com/user-attachments/assets/c59de217-7e08-48d5-9dea-e6e65a9dfc0f)

Workaround is to select per item selection all Albums (sorted by release date) and play them as action.

![Screenshot_20240831_161606_Home Assistant](https://github.com/user-attachments/assets/c2433068-69ce-49b4-949b-8f0f65801b84)
![Screenshot_20240831_161619_Home Assistant](https://github.com/user-attachments/assets/5a9b8529-77a5-4ee8-ade0-b7aa781d37ab)
![Screenshot_20240831_162730_Home Assistant](https://github.com/user-attachments/assets/8a7e5cde-28ac-4b71-998a-c558fa8b3e94)

I could not download the log file as I got an unauthorized access error when trying to download it. 
Thx
Tobias

### How to reproduce

Select an Artist with several Albums and start playing. The tracks are added in random order. 

### Music Providers

Filesystem (remote share) 
smb2 from an Synology NAS. 

### Player Providers

UPnP DLNA Providers
Denon AVR - X - 1300W

### Full log output

I can not download the log: 401: Unauthorized

### What version of Home Assistant Core are your running

2024.8.3

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by marcelveldt** [comment]
Play an entire artist is shuffled by design - not a bug. It can be changed but that is a feature request

**Reply by Panoramiac** [reply]
This is not intuitive in my opinion. This should be based on the shuffle setting. I am used to this behavior from the old time with Winamp. When this is by design then there is a simple way to play all Albums in the Album group missing.

Anyhow let's discuss how the new feature could look like. Should this be based on a setting to configure the personal preference or an enhancement of the play drop down?

**Comment by OzGav** [comment]
Moving to feature request

**Comment by trianglesis** [comment]
Adding my 5 cents.
When playing Artist Album - there should be an option to not play it in shuffle, most artists make their albums' tracks order meaningful.

The album usually starts with an introduction, lots of Live recordings too.

**Reply by marcelveldt** [reply]
Artist albums are not shuffled by default

**Reply by trianglesis** [reply]
> Artist albums are not shuffled by default

<img width="679" height="226" alt="image" src="https://github.com/user-attachments/assets/38f665cd-542e-4cd8-a5d5-93927ff7246c" />


<img width="1054" height="769" alt="Screenshot 2026-04-11 220717" src="https://github.com/user-attachments/assets/6077b646-d438-4450-b9f0-69a385a2b9c7" />

Seems they are, at least in my example, in BETA.
Or there is an option I cannot find to revert it to "default" (not shuffle), because now it always shuffles all tracks from a single album each time.

**Reply by OzGav** [reply]
I’m not getting it. I see tracks in order 1-7?

**Reply by trianglesis** [reply]
> I’m not getting it. I see tracks in order 1-7?

Yes, the view is ordered - as soon as I hit play, it starts from track 4. 

Example:
- it starts from 5th, and the queue is shuffled

<img width="2322" height="1180" alt="image" src="https://github.com/user-attachments/assets/e87a68ae-226a-416c-829b-6ad2f976c0a9" />

**Reply by OzGav** [reply]
I’ll guess, since you deleted you post, that you have shuffle turned on for the player?

**Reply by OzGav** [reply]
I see that you don’t. What player is this then?

**Reply by trianglesis** [reply]
> player

It's Snapcast, and I don't recall seeing any "shuffle" option that I may have turned on and later forgot.

<img width="1055" height="849" alt="image" src="https://github.com/user-attachments/assets/c1c352ea-723c-4c67-89a9-0cca3d58a01a" />

**Reply by OzGav** [reply]
Does it happen if you try a different player? I just tried to make sure and no the album isn’t shuffled. The only time you will see tracks shuffled without having that option turned on in the player settings is if you play an artist

**Reply by trianglesis** [reply]
> Does it happen if you try a different player? I just tried to make sure and no the album isn’t shuffled. The only time you will see tracks shuffled without having that option turned on in the player settings is if you play an artist

I tried sendspin, but I don't remember if the issue was there, unfortunately.
And it shuffles all the time. Each time I hit play, it will start from another random track, and the queue will be shuffled in a different order.

- Where can I find this shuffle option in MA? I've tried to check everything and did not find anything related to the current behaviour. I also don't think the snapcast itself (server or client) has control of track ordering.

**Reply by OzGav** [reply]
Shuffle is per player. You will need to try a different player for us to help further.

**Reply by trianglesis** [reply]
> Shuffle is per player. You will need to try a different player for us to help further.

I only have Snapcast (server) and three instances (players) for now. Will try to check later with sendspin again. 
But still, it's very surprising. I cannot see any shuffle option in the Snapcast setup in MA or at the server itself. 

Who controls the queueing and playing order? Isn't it supposed to be MA exclusively who picks and puts items in the queue and orders them?

Tested with Sendspin on Windows: plays in order and starts at track 1 each time in any album.

<img width="1434" height="763" alt="image" src="https://github.com/user-attachments/assets/4f190816-51e9-4afa-a0a1-47cff7eab24f" />

So I'm still surprised Snapcast shuffles it, where and how?

UPD: Snapcast via WEB also plays in the right order: no shuffling for an album.

<img width="1915" height="836" alt="image" src="https://github.com/user-attachments/assets/2f20fd24-84da-41cf-8a01-55c2e0e07d22" />

Where is this shuffle option?

UPD: I've completely lost the shuffle icon at the bottom of the page, at the player itself!

<img width="1074" height="125" alt="image" src="https://github.com/user-attachments/assets/5a8dc218-2899-49f2-9f62-eab1f967b456" />


I'm still cannot understand how it appeared to be ON for all three different players (by default?)

**Reply by OzGav** [reply]
Actually I was looking in the wrong place earlier. In one of your first images shuffle is clearly on 
![IMG_0510](https://github.com/user-attachments/assets/6021ce63-1416-4a4a-a263-a9d1f0a540ad)

you either tap that icon or go to the menu in the top right 
![IMG_0511](https://github.com/user-attachments/assets/83cbf478-daae-47fd-87a2-35c27cdd213f)
